### PR TITLE
Added service for get next session id

### PR DESCRIPTION
**Pull request by TayGov**
# Description

Implement service for get Next Session Id 
Clean up a few issue found along the way 

## Type of change

Please delete options that are not relevant.

- [X] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Postman test and unit tests

# Checklist:

- [X] My code follows the style guidelines of this project
- [X] I have performed a self-review of my own code
- [X] I have commented my code, particularly in hard-to-understand areas
- [X] I have made corresponding changes to the documentation
- [X] My changes generate no new warnings
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] New and existing unit tests pass locally with my changes

## For bcgov contributors:

SPDBCSC-172

### CODE REVIEW

File: src/ecrc-api/src/main/java/ca/bc/gov/open/ecrc/EcrcServices.java [hunk 1]
```diff
@@ -9,21 +9,22 @@
...
-	
-	//TODO - fill in other service method signatures here. 
-	
-}
 
+	public String getNextSessionId(String orgTicketNumber) throws EcrcServiceException;
+
+	//TODO - fill in other service method signatures here.
+
+}
```
Comment (shaunmillargov) [inline src/ecrc-api/src/main/java/ca/bc/gov/open/ecrc/EcrcServices.java]: new line here.

File: src/ecrc-api/src/main/java/ca/bc/gov/open/ecrc/EcrcServicesImpl.java [hunk 2]
```diff
@@ -41,7 +42,7 @@
 
 	private WebClient webClient = null;
 
-	private final Logger logger = LoggerFactory.getLogger(EcrcServicesImpl.class);
```
Comment (shaunmillargov) [inline src/ecrc-api/src/main/java/ca/bc/gov/open/ecrc/EcrcServicesImpl.java]: Not sure why you removed private final. This keeps it within the scope of this class only.

File: src/ecrc-api/src/main/java/ca/bc/gov/open/ecrc/configuration/EcrcProperties.java [hunk 3]
```diff
@@ -52,14 +53,15 @@ public void setGetProvincesListUri(String getProvincesListUri) {
...
-	}
+	public void setGetNextSessionIdUri(String getNextSessionIdUri) { this.getNextSessionIdUri = getNextSessionIdUri; }
+
+	public String getDoAuthenticateUserUri() { return doAuthenticateUserUri; }
+
+	public void setDoAuthenticateUserUri(String doAuthenticateUserUri) { this.doAuthenticateUserUri = doAuthenticateUserUri; }
 
-	
 
 }
```
Comment (shaunmillargov) [inline src/ecrc-api/src/main/java/ca/bc/gov/open/ecrc/configuration/EcrcProperties.java]: Newline here. .. stupid github thing..

File: src/ecrc-api/src/main/java/ca/bc/gov/open/ecrc/controller/DoAuthenticateUserController.java [hunk 4]
```diff
@@ -27,7 +27,7 @@
 		try {
 			String result = ecrcServices.doAuthenticateUser(orgTicketId);
 			if (result != null) {
-				return new ResponseEntity<>(ecrcServices.doAuthenticateUser(orgTicketId), HttpStatus.OK);
+				return new ResponseEntity<>(result, HttpStatus.OK);
```
Comment (shaunmillargov) [inline src/ecrc-api/src/main/java/ca/bc/gov/open/ecrc/controller/DoAuthenticateUserController.java]: good catch

File: src/ecrc-api/src/main/java/ca/bc/gov/open/ecrc/EcrcServicesImpl.java (hunk 2 above)
Comment (TayGov) [reply src/ecrc-api/src/main/java/ca/bc/gov/open/ecrc/EcrcServicesImpl.java]: Oh that it is intellij auto doing stuff..... I don't know why though

File: src/ecrc-api/src/main/java/ca/bc/gov/open/ecrc/configuration/EcrcProperties.java (hunk 3 above)
Comment (TayGov) [reply src/ecrc-api/src/main/java/ca/bc/gov/open/ecrc/configuration/EcrcProperties.java]: No problem it takes seconds to fix :)

**Pull request merged**